File: app/src/main/java/com/edufelip/finn/data/repositories/CommentRepository.java
package com.edufelip.finn.data.repositories;

import com.edufelip.finn.data.network.ApiService;
import com.edufelip.finn.domain.models.repositories.ICommentRepository;
import com.edufelip.finn.domain.models.Comment;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Flowable;

public class CommentRepository implements ICommentRepository {
    private final ApiService apiService;

    @Inject
    public CommentRepository(ApiService apiService) {
        this.apiService = apiService;
    }

    @Override
    public Flowable<List<Comment>> getCommentsPost(int postId) {
        return apiService.getCommentsPost(postId);
    }

    @Override
    public Flowable<Comment> saveComment(Comment comment) {
        return apiService.saveComment(comment);
    }
}

File: shared/src/commonMain/kotlin/com/edufelip/finn/shared/domain/repository/CommentRepository.kt
package com.edufelip.finn.shared.domain.repository

import com.edufelip.finn.shared.domain.model.Comment
import kotlinx.coroutines.flow.Flow

interface CommentRepository {
    fun list(postId: Int, page: Int): Flow<List<Comment>>
    fun add(postId: Int, userId: String, content: String): Flow<Comment>
}
